CHANGELOG.md

```diff
@@ -12,6 +12,12 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - Drop support for [Go 1.22]. (#6381, #6418)
 
+### Changed
+
+- Update `github.com/prometheus/client_golang` to `v1.21.1`. (#6433)
+- Set `github.com/prometheus/common/model.NameValidationScheme` to `LegacyValidation`, 
+  this avoids a breaking change introduced in [`prometheus/common@v0.62.0`](https://github.com/prometheus/common/releases/tag/v0.62.0). (#6433)
```

Comment:
```suggestion
- Update `github.com/prometheus/client_golang` to `v1.21.1`, which changes the `NameValidationScheme` to `NoEscaping`.
  This allows metrics names to keep original delimiters (e.g. `.`), rather than replacing with underscores.
  This can be reverted by setting `github.com/prometheus/common/model.NameValidationScheme` to `LegacyValidation` in `github.com/prometheus/common/model`. (#6433)
```

Content heavily inspired from https://github.com/open-telemetry/opentelemetry-go/pull/6198

Reply:
It has changed to `UTF8Validation`. In fact, `NoEscaping` is not a valid `NameValidationScheme`, but a valid `NameEscapingScheme`. I have updated the changelog accordingly.